src/sycl/fp8_scaled_mm.cpp

```diff
@@ -0,0 +1,412 @@
+  using ElementScale = cutlass::half_t;
+
+  using LayoutA = cutlass::layout::RowMajor;
+  using LayoutB = cutlass::layout::RowMajor;
+  using LayoutC = cutlass::layout::RowMajor;
+  using LayoutD = cutlass::layout::RowMajor;
+
+  using StrideScale = cute::Stride<_1, int64_t, int64_t>;
+
+  using GmemTiledCopyA = XE_2D_U8x32x32_LD_N;
+  using GmemTiledCopyB = XE_2D_U8x32x32_LD_V;
+
+  using TileShape = Shape<_256, _256, _32>;
+
+  using TiledMma = typename TiledMMAHelper<
+      MMA_Atom<XE_8x16x16_F32F16F16F32_TT>,
+      Layout<TileShape>,
+      Layout<Shape<_8, _4, _1>, Stride<_4, _1, _0>>>::TiledMMA;
+
+  static constexpr int PipelineStages = 2;
+  using GEMMDispatchPolicy = cutlass::gemm::MainloopIntelXeXMX16FP8Scaling<PipelineStages>;
+  using EpilogueDispatchPolicy = cutlass::epilogue::IntelXeXMX16;
+
+  using EpilogueOp = cutlass::epilogue::fusion::LinearCombination<
+      ElementOutput,
+      ElementComputeEpilogue,
+      ElementAccumulator,
+      ElementAccumulator,
+      cutlass::FloatRoundStyle::round_to_nearest>;
+
+  using FusionCallBacks = cutlass::epilogue::fusion::
+      FusionCallbacks<EpilogueDispatchPolicy, EpilogueOp, TileShape, decltype(tile_shape(TiledMma()))>;
+
+  // Use U16 store for FP16/BF16
+  using GmemTiledCopyStore = XE_2D_U16x8x16_ST_N;
+
+  using CollectiveEpilogue = cutlass::epilogue::collective::CollectiveEpilogue<
+      EpilogueDispatchPolicy,
+      TileShape,
+      ElementAccumulator,
+      cutlass::gemm::TagToStrideC_t<LayoutC>,
+      ElementOutput,
+      cutlass::gemm::TagToStrideC_t<LayoutD>,
+      FusionCallBacks,
+      XE_2D_U32x8x16_LD_N,
+      void,
+      void,
+      GmemTiledCopyStore,
+      void,
+      void>;
+
+  using CollectiveMainloop = cutlass::gemm::collective::CollectiveMma<
+      GEMMDispatchPolicy,
+      TileShape,
+      cute::tuple<ElementInputA, ElementScale, StrideScale>,
+      cutlass::gemm::TagToStrideA_t<LayoutA>,
+      cute::tuple<ElementInputB, ElementScale, StrideScale>,
+      cutlass::gemm::TagToStrideB_t<LayoutB>,
+      TiledMma,
+      GmemTiledCopyA,
+      void,
+      void,
+      cute::identity,
+      GmemTiledCopyB,
+      void,
+      void,
+      cute::identity>;
+
+  using GemmKernel =
+      cutlass::gemm::kernel::GemmUniversal<Shape<int, int, int, int>, CollectiveMainloop, CollectiveEpilogue>;
+
+  using Gemm = cutlass::gemm::device::GemmUniversalAdapter<GemmKernel>;
+};
+
+// Helper to check if output is FP8
+static inline bool is_fp8_dtype(at::ScalarType dtype) {
+  return dtype == at::ScalarType::Float8_e4m3fn || dtype == at::ScalarType::Float8_e5m2;
+}
+
+// Helper function to get FP8 min/max values
+static inline std::pair<float, float> get_fp8_range(at::ScalarType dtype) {
+  if (dtype == at::ScalarType::Float8_e4m3fn) {
+    // E4M3FN: max = 448, min = -448
+    return {-448.0f, 448.0f};
+  } else {
+    // Float8_e5m2
+    // E5M2: max = 57344, min = -57344
+    return {-57344.0f, 57344.0f};
+  }
+}
```

Comment:
this should have been covered in torch, aten have overloaded `std::numeric_limits`